src/helpers/deserializer.ts

```diff
@@ -0,0 +1,24 @@
+import { AxiosResponse } from 'axios';
+
+import { Resource } from 'types/resource';
+
+import { JSONObject } from './json';
+
+export type DeserializableResponse = AxiosResponse<Deserializable>;
+
+export interface Deserializable {
+  type: string;
+  id: string;
+  attributes: JSONObject;
+}
+
+export const deserialize = <T extends Resource>(data: Deserializable): T => {
+  const attributes = data.attributes;
+  const resource = {
+    id: data.id,
+    resourceType: data.type,
+    ...attributes,
+  };
```

Comment:
The `attributes` only be used once, we don't need to create a new variable for that
```suggestion
  const resource = {
    id: data.id,
    resourceType: data.type,
    ...data.attributes,
  };
```